src/Features/Core/Portable/SolutionCrawler/SolutionCrawlerProgressReporter.cs

```diff
@@ -19,109 +16,69 @@ internal partial class SolutionCrawlerRegistrationService : ISolutionCrawlerRegi
         /// </summary>
         private class SolutionCrawlerProgressReporter : ISolutionCrawlerProgressReporter
         {
-            private readonly IAsynchronousOperationListener _listener;
+            private int _progressStartCount;
+            private int _progressEvaluateCount;
 
-            // use event map and event queue so that we can guarantee snapshot and sequential ordering of events from
-            // multiple consumer from possibly multiple threads
-            private readonly SimpleTaskQueue _eventQueue;
-            private readonly EventMap _eventMap;
+            public SolutionCrawlerProgressReporter()
+            {
+                _progressStartCount = 0;
+                _progressEvaluateCount = 0;
+            }
 
-            // this is to reduce number of times we report progress on same file. this is purely for perf
-            private string _lastReportedFilePath;
+            public event EventHandler<ProgressData> ProgressChanged;
 
-            private int _count;
+            public bool InProgress => _progressStartCount > 0;
 
-            public SolutionCrawlerProgressReporter(IAsynchronousOperationListener listener)
-            {
-                _listener = listener;
-                _eventQueue = new SimpleTaskQueue(TaskScheduler.Default);
-                _eventMap = new EventMap();
+            public void Start() => ChangeProgressStatus(ref _progressStartCount, ProgressStatus.Started);
+            public void Stop() => ChangeProgressStatus(ref _progressStartCount, ProgressStatus.Stoped);
+
+            private void Evaluate() => ChangeProgressStatus(ref _progressEvaluateCount, ProgressStatus.Evaluating);
+            private void Pause() => ChangeProgressStatus(ref _progressEvaluateCount, ProgressStatus.Paused);
```

Comment:
i don't get the semantics here.  looking at this code.  you could get multiple calls to evaluate, then one call to pause, and it won't actually pause.  In that even 'pause' doesn't sound right as 'pause' generally indicates that *all* work should be stopped.  

If this really is just an increment/decrement system, and it only stops evaluating when you hit zero, then i would recommend actually calling this IncrementEvaluatingCount and DecrementEvaluatingCount.

Reply:
same here.